src/ndarray.jl

```diff
@@ -921,6 +921,25 @@ function save(filename::String, data::Dict{Base.Symbol,NDArray})
           filename, length(names), arrays, names)
 end
 
+import Base: reshape
+
+reshape{N}(arr::NDArray, dim::NTuple{N, Integer}) = _reshape(arr, dim)
+reshape{N}(arr::NDArray, dim::Vararg{Integer, N}) = _reshape(arr, dim)
+
+@inline function _reshape{N}(arr::NDArray, dim::NTuple{N, Integer})
+  h_ref = Ref{MX_handle}(C_NULL)
+  cint_dim = reverse(collect(Cint, dim))
+  @mxcall(:MXNDArrayReshape, (MX_handle, Cint, Ref{Cint}, Ref{MX_handle}),
+          arr, N, Ref(cint_dim), h_ref)
```

Comment:
I also found that there are a `reshape` operator support NDArray reshaping. what's the main difference? And... seems that (before this PR) we are using the `reshape` operator on both NDArray and SymbolicNode.

Reply:
Looking at the two implementation the `reshape` operator seems to be more flexible and we should probably use them.